pyflakes/test/test_other.py

```diff
@@ -2000,3 +2000,16 @@ def test_raise_notimplemented(self):
         self.flakes('''
         raise NotImplemented
         ''', m.RaiseNotImplemented)
+
+
+class TestAdHocConditions(TestCase):
+    def test_invalidPrintFunctions(self):
+            """
+            print keyword is invalid in Python3.x and if print_function
+            is imported from __future__ in Python2.7
+            """
+            self.flakes('''
+            from __future__ import print_function
+            import sys
+            print >>sys.stderr, "Hello"
+            ''', m.InvalidPrintSyntax)
```

Comment:
I'd advocate for more tests here, and matching the indentation of the rest of this file, e.g.,

```py
self.falkes('''
from __future__ import print_function
log = print
log("Hello")
''')

self.flakes('''
print("Hello")
''')
```